src/logging/monitor.ts

```diff
@@ -113,13 +118,17 @@ class LogMonitor {
 		onRow: (row: JournalRow) => void,
 		onError: (data: Buffer) => void,
 		onClose?: () => void,
```

Comment:
I don't see much use to keeping the `onClose` argument with the new fix. The `streamLogsFromJournal` method is set to private so we know only the `LogMonitor` class calls it

Reply:
Also, when is the the `close` event supposed to be emitted? I don't see it on the documentation https://nodejs.org/docs/latest-v12.x/api/process.html#process_process_events

Reply:
The onClose is used https://github.com/balena-io/balena-supervisor/blob/17b6b007780132898e7f6f6d9ee504e7ec6a1dbb/src/logging/monitor.ts#L142-L144 to set a follow value to true for a service. This is inside the backfill function which gets previous logs that were created while the Supervisor was not listening. This true value is used to indicate that the service can be followed now since it's been backfilled.

@codewithcheese are you able to confirm my understanding of what's going on here ?

The [onClose](https://nodejs.org/docs/latest-v12.x/api/child_process.html#child_process_event_close) event is also from the Child process class not Process object.

I wanted to make the PR as least invasive so changing events other functions used was not something I wanted to do. If we wanted to we could discuss removing onClose to use onExit as I feel they achieve the same thing.

onClose -> stream has closed but process can still be running
onExit -> process has stopped which would also cause stream to close so in effect is the same

Reply:
Thanks for clarifying that for me, I hadn't seen the backfill part of the code. It makes sense to keep the event then. I agree that it could probably work with `onExit` but is probably not super necessary